### PR TITLE
Include reconciliation to CoursierConfiguration

### PR DESCRIPTION
**Pull request by eed3si9n**
Ref https://github.com/coursier/coursier/pull/1293

This includes reconciliation into CoursierConfiguration so sbt can set it.

### CODE REVIEW

File: modules/lm-coursier/src/main/contraband/publication.json [hunk 1]
```diff
@@ -9,19 +9,29 @@
...
+          "since": "0.0.0"
         },
         {
           "name": "classifier",
-          "type": "Classifier"
+          "type": "Classifier",
+          "since": "0.0.0"
+        },
+        {
+          "name": "attributes",
```
Comment (alexarchambault) [inline modules/lm-coursier/src/main/contraband/publication.json]: Why is that extra field needed? An `Attributes` instance can be built from the `type` and `classifier` field (like you did for the default value).

File: modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala [hunk 2]
```diff
@@ -36,6 +35,27 @@ object ToCoursier {
       module.attributes
     )
 
+  def moduleMatcher(matcher: ModuleMatcher): coursier.util.ModuleMatcher =
+    if (matcher == ModuleMatcher.all) coursier.util.ModuleMatcher.all
```
Comment (alexarchambault) [inline modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala]: Do we really need that `if`? The `else` case should work in both cases.

File: modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala [hunk 3]
```diff
@@ -36,6 +35,27 @@ object ToCoursier {
...
+    else coursier.util.ModuleMatcher(module(matcher.matcher))
+
+  def moduleMatchers(matcher: ModuleMatchers): coursier.util.ModuleMatchers =
+    if (matcher == ModuleMatchers.all) coursier.util.ModuleMatchers.all
```
Comment (alexarchambault) [inline modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala]: Same as above, that `if` should be needed I think.

File: modules/lm-coursier/src/main/contraband/publication.json (hunk 1 above)
Comment (eed3si9n) [reply modules/lm-coursier/src/main/contraband/publication.json]: I thought I saw it being part of `Publication`. I'd fix this to make it same as coursier/coursier.

File: modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala (hunk 2 above)
Comment (eed3si9n) [reply modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala]: I thought I'd avoid extra allocation, but it seems like `all` is using `def` so I guess it doesn't matter.
I can remove the if part.

File: modules/lm-coursier/src/main/contraband/publication.json (hunk 1 above)
Comment (eed3si9n) [reply modules/lm-coursier/src/main/contraband/publication.json]: Addressed in 6355338735046930e5af448ef3f3a976e98c548d

File: modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala (hunk 2 above)
Comment (eed3si9n) [reply modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala]: 6355338735046930e5af448ef3f3a976e98c548d

File: modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala (hunk 3 above)
Comment (eed3si9n) [reply modules/lm-coursier/src/main/scala/lmcoursier/definitions/ToCoursier.scala]: 6355338735046930e5af448ef3f3a976e98c548d

File: modules/lm-coursier/src/main/contraband/module.json [hunk 4]
```diff
@@ -20,6 +20,46 @@
           "type": "Map[String, String]"
         }
       ]
+    },
+
+    {
+      "name": "ModuleMatcher",
```
Comment (alexarchambault) [inline modules/lm-coursier/src/main/contraband/module.json]: I'm not sure we need a `ModuleMatcher` under `lmcoursier.definitions`. Just using `lmcoursier.definitions.Module` where it's used should be enough. The less things under `lmcoursier.definitions`, the better.

Comment (eed3si9n) [reply modules/lm-coursier/src/main/contraband/module.json]: ok

Comment (eed3si9n) [reply modules/lm-coursier/src/main/contraband/module.json]: Addressed in d74df89

File: modules/sbt-coursier-shared/src/main/scala/coursier/sbtcoursiershared/InputsTasks.scala [hunk 5]
```diff
@@ -109,33 +109,33 @@ object InputsTasks {
...
     )
 
-    val attributes: Configuration => Attributes = {
+    val publications: Configuration => Publication = {
 
       val artifacts = desc.getAllDependencyArtifacts
 
       val m = artifacts.toVector.flatMap { art =>
-        val attr = Attributes(Type(art.getType), Classifier(""))
+        val pub = Publication("", Type(art.getType), Extension(""), Classifier(""))
```
Comment (alexarchambault) [inline modules/sbt-coursier-shared/src/main/scala/coursier/sbtcoursiershared/InputsTasks.scala]: Didn't you intend to use `art.getName` and `art.getExt` here?

Comment (alexarchambault) [reply modules/sbt-coursier-shared/src/main/scala/coursier/sbtcoursiershared/InputsTasks.scala]: If that's the case, do you have a case in mind where this is useful?

Comment (eed3si9n) [reply modules/sbt-coursier-shared/src/main/scala/coursier/sbtcoursiershared/InputsTasks.scala]: I think `art.getName`, `art.getExt` makes more sense.

**Pull request merged**